app/src/main/java/com/rage/clamber/Fragments/WallSectionFragment.java

```diff
@@ -1,29 +1,50 @@
 package com.rage.clamber.Fragments;
 
 
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v7.widget.DefaultItemAnimator;
 import android.support.v7.widget.GridLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import com.rage.clamber.Activities.HomePage;
 import com.rage.clamber.Adapters.WallsPageRecyclerViewAdapter;
+import com.rage.clamber.AsyncTasks.ClamberService;
+import com.rage.clamber.Data.User;
+import com.rage.clamber.Data.WallSection;
 import com.rage.clamber.R;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import butterknife.Bind;
 import butterknife.ButterKnife;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
 
 /**
  * Fragment called from WallFragment to display the individual wall sections on the selected
  * main Wall.
  */
 public class WallSectionFragment extends Fragment implements WallsPageRecyclerViewAdapter.OnWallSelectedListener {
 
-    public static final String[] WALL_NAMES = {"Wall 1", "Wall 2", "Wall 3", "Wall 4", "Wall 5","Wall 6","Wall 7","Wall 8","Wall 9", "Wall 10"};
+    public static final String TAG = WallSectionFragment.class.getSimpleName();
+    public static final String ARG_WALL_SECTION = "Wall Section Id";
+    protected User mainUser;
+    protected ArrayList<WallSection> wallSections;
```

Comment:
`List<WallSection>`, sorry to make you change it everywhere, but it's an important habit to get in